src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const copy = { ...state };
+
+  actions.forEach((elem, index) => {
+    switch (elem.type) {
+      case ('addProperties'):
+        const a = Object.assign({}, copy, elem.extraData);
+
+        Object.assign(copy, elem.extraData);
+        result.push(a);
+        break;
+
+      case ('removeProperties'):
+        elem.keysToRemove.forEach(rem => {
+          delete copy[rem];
+        });
+
+        const b = Object.assign({}, copy);
+
+        result.push(b);
+        break;
+
+      case ('clear'):
+        for (const key in copy) {
+          delete copy[key];
+        }
+
+        const c = Object.assign({}, copy);
+
+        result.push(c);
```

Comment:
You're in a loop, so push your data at the end of the switch. You don't have to do it every time